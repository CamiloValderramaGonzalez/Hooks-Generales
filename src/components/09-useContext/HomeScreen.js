import React, { useContext } from 'react'
import { UserContex } from './UserContext'

export const HomeScreen = () => {

    const { user, setUser } = useContext(UserContex);

    return (
        <div>
            <h1> HomeScreen</h1>
            <hr />
            <pre>
                {JSON.stringify(user, null, 4)}
            </pre>
        </div>
    )
}
