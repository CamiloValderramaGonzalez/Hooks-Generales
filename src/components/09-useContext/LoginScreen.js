import React, { useContext, useState } from 'react'
import { UserContex } from './UserContext'

export const LoginScreen = () => {

    const { user, setUser } = useContext(UserContex);
    const [name, setName] = useState('');
    const [email, setEmail] = useState('');

    const handleEmail = ({ target }) => {
        setEmail(target.value);
    }

    const handleName = ({ target }) => {
        setName(target.value);
    }

    return (
        <div>
            <h1> LoginScreen</h1>
            <hr />
            <div className="input-group mb-3">
                <span className="input-group-text" id="basic-addon1">@</span>
                <input
                    type="text"
                    className="form-control"
                    placeholder="Username"
                    aria-label="Username"
                    aria-describedby="basic-addon1"
                    value={name}
                    onChange={handleName}
                />
            </div>
            <div className="input-group mb-3">
                <input type="text"
                    className="form-control"
                    placeholder="email"
                    aria-label="email"
                    aria-describedby="basic-addon2"
                    value={email}
                    onChange={handleEmail}
                />
                <span className="input-group-text" id="basic-addon2">@example.com</span>
            </div>
            <button
                className='btn btn-primary'
                onClick={() => {
                    setUser({
                        name: name,
                        email: email
                    })
                }}
            >
                Guardar
            </button>
        </div>
    )
}
