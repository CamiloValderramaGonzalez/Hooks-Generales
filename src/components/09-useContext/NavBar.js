import React from 'react'
import { NavLink, Outlet } from 'react-router-dom'

export const NavBar = () => {
    return (
        <div>
            <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
                <div className="navbar-brand ms-3" to="/">useContext</div>
                <div className="container-fluid">
                    <div className="navbar-nav">
                        <NavLink className="nav-link" aria-current="page" to="/">Home</NavLink>
                        <NavLink className="nav-link" aria-current="page" to="/login">Login</NavLink>
                        <NavLink className="nav-link" aria-current="page" to="/about">About</NavLink>
                    </div>
                </div>
            </nav>
            <Outlet />
        </div>
    )
}
