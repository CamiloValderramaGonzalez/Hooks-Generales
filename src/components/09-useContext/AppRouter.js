import React from "react";
import {
    BrowserRouter,
    Routes,
    Route,
} from "react-router-dom";

import { NavBar } from "./NavBar";

import { AboutScreen } from "./AboutScreen";
import { HomeScreen } from "./HomeScreen";
import { LoginScreen } from "./LoginScreen";

export const AppRouter = () => {
    return (
        <BrowserRouter>
            <NavBar />
            <div className='container mt-4'>
                <Routes >
                    <Route path="/">
                        <Route index element={<HomeScreen />} />
                        <Route path="about" element={<AboutScreen />} />
                        <Route path="login" element={<LoginScreen />} />
                        <Route path="*" element={<HomeScreen />} />
                    </Route>
                </Routes>
            </div>
        </BrowserRouter>
    )
}
