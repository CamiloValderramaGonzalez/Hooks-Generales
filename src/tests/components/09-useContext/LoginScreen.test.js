import React from 'react';
import { mount } from 'enzyme'
import { UserContex } from '../../../components/09-useContext/UserContext';
import { LoginScreen } from '../../../components/09-useContext/LoginScreen';

describe('Pruebas en <LoginScreen />', () => {

    const user = {
        name: 'Camilo',
        email: 'a@a.com'
    }

    const setUser = jest.fn();

    const wrapper = mount(
        <UserContex.Provider value={{
            user,
            setUser
        }}>

            <LoginScreen />
        </UserContex.Provider>
    );

    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe ejecutarse el setUSer con el argumento esperado', () => {
        wrapper.find('button').prop('onClick')();
        expect(setUser).toHaveBeenCalledWith({ "email": "", "name": "" })
    })
})