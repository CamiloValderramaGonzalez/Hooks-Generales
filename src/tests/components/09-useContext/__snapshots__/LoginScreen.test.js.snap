// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen /> Debe mostrarse correctamente 1`] = `
<div>
  <h1>
     LoginScreen
  </h1>
  <hr />
  <div
    className="input-group mb-3"
  >
    <span
      className="input-group-text"
      id="basic-addon1"
    >
      @
    </span>
    <input
      aria-describedby="basic-addon1"
      aria-label="Username"
      className="form-control"
      onChange={[Function]}
      placeholder="Username"
      type="text"
      value=""
    />
  </div>
  <div
    className="input-group mb-3"
  >
    <input
      aria-describedby="basic-addon2"
      aria-label="email"
      className="form-control"
      onChange={[Function]}
      placeholder="email"
      type="text"
      value=""
    />
    <span
      className="input-group-text"
      id="basic-addon2"
    >
      @example.com
    </span>
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Guardar
  </button>
</div>
`;
